pkg/provider/provider.go

```diff
@@ -283,15 +285,22 @@ func (p *Provider) MountSecretsStoreObjectContent(ctx context.Context, attrib ma
 
 		objectContent, err := getContentBytes(content, keyVaultObject.ObjectType, keyVaultObject.ObjectEncoding)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
-		if err := os.WriteFile(filepath.Join(targetPath, fileName), objectContent, permission); err != nil {
-			return nil, errors.Wrapf(err, "failed to write file %s at %s", fileName, targetPath)
+		// if the feature to return secrets to CSI driver isn't enabled, the provider will continue to write
+		// the contents to the filesystem.
+		if !*DriverWriteSecrets {
+			if err := os.WriteFile(filepath.Join(targetPath, fileName), objectContent, permission); err != nil {
+				return nil, nil, errors.Wrapf(err, "failed to write file %s at %s", fileName, targetPath)
+			}
+			klog.InfoS("successfully wrote file", "file", fileName, "pod", klog.ObjectRef{Namespace: p.PodNamespace, Name: p.PodName})
+		} else {
+			// these files will returned to the CSI driver as part of grpc response
+			files[fileName] = objectContent
```

Comment:
should we still log success here? would be useful when we debug issues between driver and provider.

Reply:
Done!